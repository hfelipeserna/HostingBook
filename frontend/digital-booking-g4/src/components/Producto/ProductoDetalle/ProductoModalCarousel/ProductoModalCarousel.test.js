import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import ProductoModaCarousel from "./ProductoModalCarousel"

describe("Producto Modal Carousel Test", function () {
    let estaAbierto = true

    const onCloseRequest = () => {

    }
    const imagenes = [{
        id: 1,
        imagenUr: " esta es la imagen url",
        imagenTitulo: " este es el titulo de la imagen"
    }]
    test("no renderiza en false", () => {
        estaAbierto = false
        const renderizacionsimple = render(<ProductoModaCarousel estaAbierto={estaAbierto}
            onCloseRequest={onCloseRequest}
            imagenes={imagenes}
        ></ProductoModaCarousel>)
        expect(renderizacionsimple).toMatchSnapshot()
    })
    test("Renderiza Por default", () => {
        estaAbierto = true
        const renderizacionsimple = render(<ProductoModaCarousel estaAbierto={estaAbierto}
            onCloseRequest={onCloseRequest}
            imagenes={imagenes}
        ></ProductoModaCarousel>)
        expect(renderizacionsimple).toMatchSnapshot()

    })
})