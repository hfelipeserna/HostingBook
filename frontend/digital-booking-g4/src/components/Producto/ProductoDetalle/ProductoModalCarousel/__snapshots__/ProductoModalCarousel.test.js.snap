// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Producto Modal Carousel Test Renderiza Por default 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="background"
      >
        <section
          class="background"
        >
          <div
            class="modal"
          >
            <span
              class="close"
            >
              X
            </span>
            <div
              class="carousel-root"
              tabindex="0"
            >
              <div
                class="carousel carousel-slider"
                style="width: 100%;"
              >
                <ul
                  class="control-dots"
                >
                  <li
                    aria-label="slide item 1"
                    class="dot selected"
                    role="button"
                    tabindex="0"
                    value="0"
                  />
                </ul>
                <div
                  class="slider-wrapper axis-horizontal"
                  style="height: auto;"
                >
                  <ul
                    class="slider animated"
                    style="transform: translate3d(-100%,0,0); transition-duration: 750ms;"
                  >
                    <li
                      class="slide selected previous"
                    >
                      <div
                        class="imagen"
                      >
                        <img
                          alt=" este es el titulo de la imagen"
                        />
                        <p
                          class="legend"
                        >
                           este es el titulo de la imagen
                        </p>
                      </div>
                    </li>
                    <li
                      class="slide selected previous"
                    >
                      <div
                        class="imagen"
                      >
                        <img
                          alt=" este es el titulo de la imagen"
                        />
                        <p
                          class="legend"
                        >
                           este es el titulo de la imagen
                        </p>
                      </div>
                    </li>
                  </ul>
                </div>
              </div>
              <div
                class="carousel"
              >
                <div
                  class="thumbs-wrapper axis-vertical"
                >
                  <button
                    aria-label="previous slide / item"
                    class="control-arrow control-prev control-disabled"
                    type="button"
                  />
                  <ul
                    class="thumbs animated"
                    style="transform: translate3d(0,0,0); transition-duration: 750ms;"
                  >
                    <li
                      aria-label="slide item 1"
                      class="thumb selected"
                      role="button"
                      style="width: 80px;"
                      tabindex="0"
                    >
                      <img
                        alt=" este es el titulo de la imagen"
                      />
                    </li>
                  </ul>
                  <button
                    aria-label="next slide / item"
                    class="control-arrow control-next"
                    type="button"
                  />
                </div>
              </div>
            </div>
          </div>
        </section>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="background"
    >
      <section
        class="background"
      >
        <div
          class="modal"
        >
          <span
            class="close"
          >
            X
          </span>
          <div
            class="carousel-root"
            tabindex="0"
          >
            <div
              class="carousel carousel-slider"
              style="width: 100%;"
            >
              <ul
                class="control-dots"
              >
                <li
                  aria-label="slide item 1"
                  class="dot selected"
                  role="button"
                  tabindex="0"
                  value="0"
                />
              </ul>
              <div
                class="slider-wrapper axis-horizontal"
                style="height: auto;"
              >
                <ul
                  class="slider animated"
                  style="transform: translate3d(-100%,0,0); transition-duration: 750ms;"
                >
                  <li
                    class="slide selected previous"
                  >
                    <div
                      class="imagen"
                    >
                      <img
                        alt=" este es el titulo de la imagen"
                      />
                      <p
                        class="legend"
                      >
                         este es el titulo de la imagen
                      </p>
                    </div>
                  </li>
                  <li
                    class="slide selected previous"
                  >
                    <div
                      class="imagen"
                    >
                      <img
                        alt=" este es el titulo de la imagen"
                      />
                      <p
                        class="legend"
                      >
                         este es el titulo de la imagen
                      </p>
                    </div>
                  </li>
                </ul>
              </div>
            </div>
            <div
              class="carousel"
            >
              <div
                class="thumbs-wrapper axis-vertical"
              >
                <button
                  aria-label="previous slide / item"
                  class="control-arrow control-prev control-disabled"
                  type="button"
                />
                <ul
                  class="thumbs animated"
                  style="transform: translate3d(0,0,0); transition-duration: 750ms;"
                >
                  <li
                    aria-label="slide item 1"
                    class="thumb selected"
                    role="button"
                    style="width: 80px;"
                    tabindex="0"
                  >
                    <img
                      alt=" este es el titulo de la imagen"
                    />
                  </li>
                </ul>
                <button
                  aria-label="next slide / item"
                  class="control-arrow control-next"
                  type="button"
                />
              </div>
            </div>
          </div>
        </div>
      </section>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Producto Modal Carousel Test no renderiza en false 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
